test/ec_driver.cpp

```diff
@@ -78,23 +79,38 @@ void create_coding_files(
     std::vector<quadiron::Properties> c_props(fec->n_outputs);
 
     for (unsigned i = 0; i < fec->n_data; i++) {
-        snprintf(filename, sizeof(filename), "%s.d%d", prefix, i);
+        snprintf(filename, sizeof(filename), "%s.d%0*d", prefix, zpad, i);
         if (vflag)
             std::cerr << "create: opening data " << filename << "\n";
-        d_files[i] = new std::ifstream(filename);
+        try {
+          d_files[i] = new std::ifstream(filename);
+        } catch (const std::ifstream::failure& e) {
+          std::cerr << "Exception opening " << filename << "\n";
+          exit(1);
```

Comment:
`std::exit(EXIT_FAILURE);` would be cleaner/more explicit then `exit(1)` (may need an `#include <cstdlib>`).

Same remarks for the other occurrences.